src/Form/FormAnalyzer.js

```diff
@@ -223,6 +223,18 @@ class FormAnalyzer {
         });
     }
 
+    /**
+     * Checks if the element is present in the cusotm elements registry and ends with a '-link' suffix.
+     * If it does, it checks if it contains an anchor element inside.
+     * @param {any} el
+     * @returns
+     */
+    isCustomWebElementLink(el) {
+        const tagName = el.nodeName.toLowerCase();
+        const isCustomElement = customElements != null && customElements.get(tagName) != null;
+        return isCustomElement && /-link$/.test(tagName) && findElementsInShadowTree(el, 'a').length > 0;
+    }
+
```

Comment:
Tried to add a check for some extra help in scoring, if the link is a custom web component. In case of reddit it's `auth-flow-link` and then we have an anchor element within it. I think it's quite a narrow optimization but possibly a pattern for slotted elements.

Reply:
Haven't actually checked, but could this

```js
customElements != null && customElements.get(tagName) != null;
```

be

```js
customElements?.get(tagName) != null;
```

Reply:
Actually `customElements.get(tagName) != null;`  is enough. `customElements` is always avaiable, `get` acts normal and returns undefined when nothing's there.

Reply:
NVM - not enough 😄  Tests fail because of it 😄 So going with your suggestion :)